### PR TITLE
chore: add commands for liquidity provision cancellation and amendment

### LINKED ISSUE #205: Update Liquidity Provision Commands
# Task Overview

To complete https://github.com/vegaprotocol/vega/issues/4561
Update the protobuf definitions and add validation

### PR DESCRIPTION
**Pull request by guoguojin**
Close #205

### CODE REVIEW

File: commands/liquidity_provision_submission_test.go [hunk 1]
```diff
@@ -182,3 +184,225 @@ func TestLiquidityProvisionSubmission(t *testing.T) {
...
+						{Reference: types.PeggedReference_PEGGED_REFERENCE_BEST_ASK, Offset: 10, Proportion: 1},
+					},
+					Buys: []*types.LiquidityOrder{},
+				},
+			},
+			wantErr:   assert.Error,
+			errString: "liquidity_provision_submission.buys (empty shape)",
+		},
+		{
+			name: "Should return an error if buy side shape is provided with no sell side shape",
```
Comment (jeremyletang) [inline commands/liquidity_provision_submission_test.go:346]: This should not be an issue for an amendment? That would suggest that no amendment are done on the sell side and it stay as it was.

Comment (jeremyletang) [reply commands/liquidity_provision_submission_test.go:346]: Or maybe that differenciate between an empty slice and nil? Not sure that proto would do any differenciation at this point between a nil and empty slice.

Comment (guoguojin) [reply commands/liquidity_provision_submission_test.go:346]: Yes, sorry, you are right. This validation is required on submission, but not on validation as an empty/nil slice means no changes to the existing submission. I will correct this.

File: sources/vega/commands/v1/commands.proto [hunk 2]
```diff
@@ -92,6 +92,25 @@ message LiquidityProvisionSubmission {
...
+  string id = 1; // this is required
+  string market_id = 2; // this can be remove if we get a nice way to get provisions IDs to market
+}
+
+// Amend a liquidity provision request
+message LiquidityProvisionAmendment {
+  string id = 1; // this is required
+  string market_id = 2; // this can be remove if we get a nice way to get provisions IDs to market
+
+  // From here at least one of the following is required to consider the command valid
```
Comment (wwestgarth) [inline sources/vega/commands/v1/commands.proto:106]: Should this comment be reflected in `checkLiquidityProvisionAmendment` will that be in `core`?

Comment (guoguojin) [reply sources/vega/commands/v1/commands.proto:106]: I'm sorry, I don't know what you mean. If you are referring to line 106, I have just realised that the validation doesn't check for this so I need to add it.

**Comment by guoguojin** [comment]
I've updated more protos that touch LiquidityProvisionSubmission in the wallet api and events protos, so think it best if it was re-reviewed just in case.

**Review by wwestgarth — APPROVED** [review]

**Pull request merged**